src/main/java/org/fenixedu/academictreasury/services/reports/DocumentPrinter.java

```diff
@@ -35,99 +35,35 @@
-import org.fenixedu.treasury.services.reports.helpers.DateHelper;
-import org.fenixedu.treasury.services.reports.helpers.EnumerationHelper;
-import org.fenixedu.treasury.services.reports.helpers.LanguageHelper;
-import org.fenixedu.treasury.services.reports.helpers.MoneyHelper;
-import org.fenixedu.treasury.services.reports.helpers.NumbersHelper;
-import org.fenixedu.treasury.services.reports.helpers.StringsHelper;
-import org.joda.time.LocalDate;
-
-import com.google.common.base.Strings;
-import com.qubit.terra.docs.core.DocumentGenerator;
-import com.qubit.terra.docs.core.DocumentTemplateEngine;
-import com.qubit.terra.docs.core.IDocumentTemplateService;
 
 public class DocumentPrinter {
-    private static final String TEMPLATES_TUITIONS_PAYMENT_PLAN = "templates/tuitionsPaymentPlan.odt";
-
-    static {
-        registerService();
-    }
+    private static AcademicDocumentPrinterInterface SINGLETON;
 
-    private static void registerService() {
-        IDocumentTemplateService service = new DocumentPrinterConfiguration();
-        DocumentTemplateEngine.registerServiceImplementations(service);
+    public static void registerPrinterInterface(AcademicDocumentPrinterInterface printer) {
+        SINGLETON = printer;
+        SINGLETON.init();
     }
 
-    public static final String PDF = DocumentGenerator.PDF;
-    public static final String ODT = DocumentGenerator.ODT;
-
-    private static void registerHelpers(DocumentGenerator generator) {
-        generator.registerHelper("dates", new DateHelper());
-        generator.registerHelper("lang", new LanguageHelper());
-        generator.registerHelper("numbers", new NumbersHelper());
-        generator.registerHelper("enumeration", new EnumerationHelper());
-        generator.registerHelper("strings", new StringsHelper());
-        generator.registerHelper("money", new MoneyHelper());
-    }
+    public static final String PDF = org.fenixedu.treasury.services.reports.DocumentPrinter.PDF;
+    public static final String ODT = org.fenixedu.treasury.services.reports.DocumentPrinter.ODT;
 
     //https://github.com/qub-it/fenixedu-qubdocs-reports/blob/master/src/main/java/org/fenixedu/academic/util/report/DocumentPrinter.java
     public static byte[] printRegistrationTuititionPaymentPlan(Registration registration, String outputMimeType) {
-        IAcademicTreasuryPlatformDependentServices services = AcademicTreasuryPlataformDependentServicesFactory.implementation();
-        
-        final Person p = registration.getStudent().getPerson();
-        final String addressFiscalCountryCode = PersonCustomer.addressCountryCode(p);
-        final String fiscalNumber = PersonCustomer.fiscalNumber(p);
-        if (Strings.isNullOrEmpty(addressFiscalCountryCode) || Strings.isNullOrEmpty(fiscalNumber)) {
-            throw new AcademicTreasuryDomainException("error.PersonCustomer.fiscalInformation.required");
+        if(SINGLETON != null) {
```

Comment:
@joaofreitas-qubit the same comment I made in the `fenixedu-treasury` I can make here, can refactor this into a more functional way. But like I said before, this is non blocking